src/app/zap-templates/templates/app/cluster-enums.zapt

```diff
@@ -28,8 +28,9 @@ k{{asUpperCamelCase label}} = {{asHex value 2}},
 {{#if (isWeaklyTypedEnum label)}}
 #else // CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
 using {{asType label}} = EmberAf{{asType label}};
-#endif
+#endif // CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
 {{/if}}
+static {{asType label}} __attribute__((unused)) k{{asType label}}FirstUnusedEnumVal = static_cast<{{asType label}}>({{first_unused_enum_value mode="first_unused"}});
```

Comment:
Shouldn't this be in the enum class for the cases when we have an enum class?

Reply:
then I have to modify more server logic to explicitly process these new enum value, for example
../../src/app/clusters/diagnostic-logs-server/diagnostic-logs-server.cpp:41:21: error: enumeration value 'kFirstUnusedEnumVal' not handled in switch [-Werror,-Wswitch]
            switch (payload.intent)
                    ^~~~~~~~~~~~~~
../../src/app/CommandHandlerInterface.h:223:13: note: in instantiation of function template specialization 'DiagnosticLogsCommandHandler::InvokeCommand(chip::app::CommandHandlerInterface::HandlerContext &)::(anonymous class)::operator()<chip::app::CommandHandlerInterface::HandlerContext, chip::app::Clusters::DiagnosticLogs::Commands::RetrieveLogsRequest::DecodableType>' requested here
            func(handlerContext, requestPayload);

Reply:
As discussed on Slack, that is _exactly_ the point of adding this value: to help catch all the places that need to be checking for it.

Also as discussed, to keep scope sane for this PR we can add the value inside the enum class commented out, and then do a separate PR that uncomments it and fixes up all the places that catches.

Reply:
Created https://github.com/project-chip/connectedhomeip/pull/20907 to handle the followup.